plugin.py

```diff
@@ -0,0 +1,92 @@
+from LSP.plugin import AbstractPlugin, ClientConfig, register_plugin, unregister_plugin, WorkspaceFolder
+from LSP.plugin.core.typing import Dict, Optional, List
+from shutil import which
+import os
+import sublime
+import urllib.request
+
+
+MARKSMAN_TAG = '2022-08-27'
+MARKSMAN_RELEASES_BASE = 'https://github.com/artempyanykh/marksman/releases/download/{tag}/{platform}'
+USER_AGENT = 'Sublime Text LSP'
+
+
+def marksman_binary() -> Optional[str]:
+    platform = sublime.platform()
+    if platform == 'osx':
+        return 'marksman-macos'
+    if platform == 'windows':
+        return 'marksman.exe'
+    if platform == 'linux':
+        return 'marksman-linux'
+    return None
+
+
+class Marksman(AbstractPlugin):
+    @classmethod
+    def name(cls) -> str:
+        return 'marksman'
+
+    @classmethod
+    def basedir(cls) -> str:
+        return os.path.join(cls.storage_path(), __package__)
+
+    @classmethod
+    def marksman_path(cls) -> str:
+        return os.path.join(cls.basedir(), 'bin', marksman_binary() or 'unsupported_platform')
+
+    @classmethod
+    def additional_variables(cls) -> Optional[Dict[str, str]]:
+        return {
+            'marksman_bin': marksman_binary() or 'unsupported_platform'
+        }
+
+    @classmethod
+    def server_version(cls) -> str:
+        return MARKSMAN_TAG
+
+    @classmethod
+    def current_server_version(cls) -> Optional[str]:
+        try:
+            with open(os.path.join(cls.basedir(), 'VERSION'), 'r') as fp:
+                return fp.read()
+        except Exception:
+            return None
+
+    @classmethod
+    def can_start(cls, window: sublime.Window, initiating_view: sublime.View,
+                  workspace_folders: List[WorkspaceFolder], configuration: ClientConfig) -> Optional[str]:
+        if marksman_binary() is None:
+            return 'Platform "{}" is not supported'.format(sublime.platform())
```

Comment:
Hmm, this runs after `install_or_update` so might have to move it back to `needs_update_or_installation`.